metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/eosc/AccountingApiMetricsPushService.java

```diff
@@ -0,0 +1,125 @@
+package eu.openaire.mas.eosc;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import eu.eosc.accounting_system.client.api.MetricApi;
+import eu.eosc.accounting_system.client.invoker.ApiClient;
+import eu.eosc.accounting_system.client.invoker.auth.Authentication;
+import eu.eosc.accounting_system.client.invoker.auth.OAuth;
+import eu.eosc.accounting_system.client.model.InlineResponse2011;
+import eu.eosc.accounting_system.client.model.MetricRequest;
+import eu.openaire.mas.delivery.MetricMetadata;
+import eu.openaire.mas.delivery.provider.MetricsMetadataProvider;
+import eu.openaire.mas.delivery.provider.MetricsProvider;
+
+/**
+ * A service which regularly pushes metric values to an EOSC accounting API instance.
+ */
+@Service
+public class AccountingApiMetricsPushService {
+
+    private static Logger log = LoggerFactory.getLogger(AccountingApiMetricsPushService.class);
+
+    @Value("${eosc.accounting-api.base-path}")
+    private String basePath;
+
+    @Value("${counter.range.start}")
```

Comment:
It will be nice to describe it, at least briefly, what format this value should be provided in.

Reply:
And I don't see this value specified in a properties file.

Reply:
This property is not specific to the accounting system integration, it is older and it is in `application.properties`.